File: app/src/main/java/com/example/fruitsdiary/usecase/addeditentry/AddEditEntryManager.java
package com.example.fruitsdiary.usecase.addeditentry;

import android.support.annotation.NonNull;

import com.example.fruitsdiary.model.FruitEntry;

interface AddEditEntryManager {

    void saveEntry();

    void addOrUpdateFruitEntry(FruitEntry fruitEntry);

    void deleteFruitEntry(@NonNull FruitEntry fruitEntry);

    void deleteEntry();

    boolean contains(FruitEntry fruitEntry);

    FruitEntry getCorrespondingFruitEntry(FruitEntry fruitEntry);

    void showOverlay();

    void hideOverlay();
}

File: app/src/main/java/com/example/fruitsdiary/usecase/addeditentry/AddEditEntryManager.kt
package com.example.fruitsdiary.usecase.addeditentry

import com.example.fruitsdiary.model.FruitEntry

interface AddEditEntryManager {

    fun saveEntry()

    fun addOrUpdateFruitEntry(fruitEntry: FruitEntry)

    fun deleteFruitEntry(fruitEntry: FruitEntry)

    fun deleteEntry()

    operator fun contains(fruitEntry: FruitEntry): Boolean

    fun getCorrespondingFruitEntry(fruitEntry: FruitEntry): FruitEntry?

    fun showOverlay()

    fun hideOverlay()
}
